src/compiler/base/declare-variable/sequencetype-with-uqnames.xsl

```diff
@@ -15,7 +15,8 @@
    -->
    <xsl:function name="x:lexical-to-UQName-in-sequence-type" as="xs:string?">
       <xsl:param name="context" as="element()"/>
-      <xsl:variable name="as-attribute-value" as="xs:string?" select="$context/@as"/>
+      <xsl:param name="attribute-local-name" as="xs:string"/>
```

Comment:
I generalized the function from #2103 so it can apply to either `@as` (relevant for #2103) or `@result-type` (relevant here).